pyscriptjs/src/components/pyconfig.ts

```diff
@@ -27,25 +41,65 @@ export class PyConfig extends BaseEvalElement {
         super();
     }
 
-    connectedCallback() {
+    extractFromSrc() {
+        return validateConfig(readTextFromPath(this.getAttribute('src')));
+    }
+
+    extractFromInline() {
         this.code = this.innerHTML;
         this.innerHTML = '';
+        return validateConfig(this.code);
+    }
+
+    getDefaultConfig()
+    {
+        return inJest() ? defaultConfig : JSON.parse(defaultConfig);
```

Comment:
Can you clarify more about this and why we need this? It's usually not a good practice to have conditional code checking whether we are in prod or test env. Can't we handle things on the test context side? If really really not, can you add the reasons also as a comment in the code, for awareness and posterity?

Reply:
So, we are importing external files such as `import pyscript from './python/pyscript.py';` and `import defaultConfig from '../pyscript.json';`. Both of them are not JS files and hence need to be converted to text. So the following snippet is used in `rollup.config.js`

```
string({
      include: "./src/**/*.py",
}),
string({
      include: "./src/**/*.json",
})
```

Thus we use `JSON.parse(defaultConfig)`.
However, for reasons unknown to me (need some JS ninja help) -- when I run unit tests in `jest` with `make test-ts` -- 
The import is actually a JSON object and not a string, thus the check here.

I tried exploring `https://www.npmjs.com/package/@rollup/plugin-json` but then:
- I couldn't make it work
- I want to avoid using libraries so as to avoid increasing shipping overheads

Reply:
"I want to avoid using libraries so as to avoid increasing shipping overheads" 😍 

Within .tsconfig add "resolveJsonModule":true
This results in JSON conversion to JSObjects at compile-time and avoids JSON.parse. You can then remove 
```
string({
      include: "./src/**/*.json",
})
```

Reply:
Thanks, while the editor stop complaining, it still fails on `npm run build` with the following:
```
Error: Unexpected token (Note that you need @rollup/plugin-json to import JSON files)
```

I guess the `rollup` thingie is still failing at importing this.

Reply:
Ok so, I can remove:
```
string({
      include: "./src/**/*.json",
})
```

but we need both `"resolveJsonModule":true` and the library `https://www.npmjs.com/package/@rollup/plugin-json` to make it work.

so if we need to remove the `inJest` stuff, we need this additional library aka a bit of shipping overhead.

Reply:
done, me and @tedpatrick decided on not to import files, so as to avoid dealing with plugins etc.

Reply:
This probably should be a more explicit discussion somewhere but I think we should be moving all the `py` code of  the JS package and loading it separately (mostly all) as a package.